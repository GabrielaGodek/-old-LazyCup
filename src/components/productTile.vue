<script>
import { mapStores, mapWritableState, mapActions } from 'pinia'
import { useOrdersStore } from '@/store/orders'

export default {
  name: 'productTile',
  props: {
    coffee: {
      type: Object,
      required: true
    }
  },
  computed: {
    ...mapWritableState(useOrdersStore, ['coffees']),
    ...mapStores(useOrdersStore)
  },
  methods: {
    ...mapActions(useOrdersStore, ['addItem']),
    addToFav(coffee) {
      let item = (this.ordersStore.coffees.filter(el => el._id === coffee._id))
      item[0].isFav === true ? (item[0].isFav = false) : (item[0].isFav = true)
    },
    basketShort(cartItem) {
      this.addItem({
        id: cartItem._id,
        amount: 1,
        name: cartItem.name,
        price: cartItem.price,
        salePrice: cartItem.salePrice,
        image: cartItem.image,
        date: new Date().toJSON().slice(0, 10).replace(/-/g, '/')
      })
      // console.log(this.ordersStore.orders)
      this.$router.push({ name: 'cart' })
    }
  }
}
</script>

<template>
  <div class="product_tile">
    <div class="tile">
      <div class="action">
        <div class="cart">
          <font-awesome-icon
            icon="fa-solid fa-cart-shopping"
            size="lg"
            style="color: #252525"
            @click="basketShort(coffee)"
          />
        </div>
        <div class="fav" @click="addToFav(coffee)">
          <template v-if="coffee.isFav && coffee.isFav === true">
            <font-awesome-icon icon="fa-solid fa-heart" size="lg" style="color: #252525" />
          </template>
          <template v-else>
            <font-awesome-icon icon="fa-regular fa-heart" size="lg" style="color: #252525" />
          </template>
        </div>
      </div>
      <div
        class="image"
        :id="coffee._id"
        @click="this.$router.push({ name: 'coffee', params: { id: coffee._id } })"
      >
        <img :src="'https://link to domain/' + coffee.image" :alt="coffee.name" />
      </div>
      <div
        class="description"
        @click="this.$router.push({ name: 'coffee', params: { id: coffee._id } })"
        :id="coffee._id"
      >
        <h1 class="title">
          {{ coffee.name }}
        </h1>
        <p class="details">
          {{ coffee.description }}
        </p>
        <div class="price_container">
          <h2 class="new_price" v-if="coffee.salePrice">{{ coffee.salePrice }} zł</h2>
          <span class="old_price"> {{ coffee.price }} zł</span>
        </div>
      </div>
    </div>
  </div>
</template>

<style scoped>
.listing .product_tile .details {
  display: none;
}
</style>
