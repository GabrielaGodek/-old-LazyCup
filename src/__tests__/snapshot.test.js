import { it, describe, expect } from 'vitest'
import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import productTile from '@/components/productTile.vue'

describe('snapshot productTile.vue', () => {
    it('renders correctly', () => {
        const coffee = {
            _id: '651964f924f0811dee90bfed',
            name: 'Espresso',
            description: 'A rich and flavorful coffee.',
            price: 12,
            salePrice: 10,
            image: 'Espresso.svg',
        };

        const wrapper = shallowMount(productTile, {
            props: {
                coffee: coffee,
            },
            global: {
                components: {
                    'router-link': RouterLinkStub
                }
            }
        })

        expect(wrapper.element).toMatchSnapshot()
    })
})
