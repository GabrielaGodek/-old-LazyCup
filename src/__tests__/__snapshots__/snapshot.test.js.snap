// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot productTile.vue > renders correctly 1`] = `
<div
  class="product_tile"
  data-v-27b69f7b=""
  id="651964f924f0811dee90bfed"
>
  <div
    class="tile"
    data-v-27b69f7b=""
  >
    <div
      class="action"
      data-v-27b69f7b=""
    >
      <div
        class="cart"
        data-v-27b69f7b=""
      >
        <font-awesome-icon
          data-v-27b69f7b=""
          icon="fa-solid fa-cart-shopping"
          size="lg"
          style="color: rgb(37, 37, 37);"
        />
      </div>
      <div
        class="fav"
        data-v-27b69f7b=""
      >
        <font-awesome-icon
          data-v-27b69f7b=""
          icon="fa-regular fa-heart"
          size="lg"
          style="color: rgb(37, 37, 37);"
        />
      </div>
    </div>
    <div
      class="image"
      data-v-27b69f7b=""
      id="651964f924f0811dee90bfed"
    >
      <img
        alt="Espresso"
        data-v-27b69f7b=""
        lazy-loading=""
        src="/coffees_icons/Espresso.svg"
      />
    </div>
    <div
      class="description"
      data-v-27b69f7b=""
      id="651964f924f0811dee90bfed"
    >
      <h1
        class="title"
        data-v-27b69f7b=""
      >
        Espresso
      </h1>
      <p
        class="details"
        data-v-27b69f7b=""
      >
        A rich and flavorful coffee.
      </p>
      <div
        class="price_container"
        data-v-27b69f7b=""
      >
        <h4
          class="new_price"
          data-v-27b69f7b=""
        >
          10 zł 
        </h4>
        <h5
          class="old_price"
          data-v-27b69f7b=""
        >
          12 zł
        </h5>
      </div>
    </div>
  </div>
</div>
`;
